Fix misuages for TORCH_CHECK/TORCH_INTERNAL_ASSERT with string

dzhulgakov opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26897 Fix misuages for TORCH_CHECK/TORCH_INTERNAL_ASSERT with string**
* #26840 Remove fbgemm_is_cpu_supported in favor of torch.backends.quantized.supported_qengines

TORCH_INTERNAL_ASSERT("foo") doesn't do what you think it does :)

I'll try to do a fix to catch it in the compiler, but for now - let's fix usages

Found them using regex:
```
ag --cpp "TORCH_(CHECK|INTERNAL_ASSERT)\([ \n]*\"" --multiline
```

Differential Revision: [D17624299](https://our.internmc.facebook.com/intern/diff/D17624299)

Commit:
Fix misuages for TORCH_CHECK/TORCH_INTERNAL_ASSERT with string

[ghstack-poisoned]

smessmer reviewed (approved):
wait what? These can't have worked...

Commit:
Update on "Fix misuages for TORCH_CHECK/TORCH_INTERNAL_ASSERT with string"


TORCH_INTERNAL_ASSERT("foo") doesn't do what you think it does :)

I'll try to do a fix to catch it in the compiler, but for now - let's fix usages

Found them using regex:
```
ag --cpp "TORCH_(CHECK|INTERNAL_ASSERT)\([ \n]*\"" --multiline
```

[ghstack-poisoned]

Commit:
Update on "Fix misuages for TORCH_CHECK/TORCH_INTERNAL_ASSERT with string"


TORCH_INTERNAL_ASSERT("foo") doesn't do what you think it does :)

I'll try to do a fix to catch it in the compiler, but for now - let's fix usages

Found them using regex:
```
ag --cpp "TORCH_(CHECK|INTERNAL_ASSERT)\([ \n]*\"" --multiline
```

[ghstack-poisoned]

Commit:
Update on "Fix misuages for TORCH_CHECK/TORCH_INTERNAL_ASSERT with string"


TORCH_INTERNAL_ASSERT("foo") doesn't do what you think it does :)

I'll try to do a fix to catch it in the compiler, but for now - let's fix usages

Found them using regex:
```
ag --cpp "TORCH_(CHECK|INTERNAL_ASSERT)\([ \n]*\"" --multiline
```

Differential Revision: [D17624299](https://our.internmc.facebook.com/intern/diff/D17624299)

[ghstack-poisoned]

Commit:
Update on "Fix misuages for TORCH_CHECK/TORCH_INTERNAL_ASSERT with string"


TORCH_INTERNAL_ASSERT("foo") doesn't do what you think it does :)

I'll try to do a fix to catch it in the compiler, but for now - let's fix usages

Found them using regex:
```
ag --cpp "TORCH_(CHECK|INTERNAL_ASSERT)\([ \n]*\"" --multiline
```

Differential Revision: [D17624299](https://our.internmc.facebook.com/intern/diff/D17624299)

[ghstack-poisoned]
